src/domain/event/similarEvents/__tests__/SimilarEvents.test.tsx

```diff
@@ -83,3 +88,32 @@ test('should render similar event cards', async () => {
     ).toBeInTheDocument();
   });
 });
+
+it('has return path on similar event link', async () => {
+  const path = ROUTES.EVENT;
+  const route = path.replace(':id', event.id);
+  const { history } = render(<SimilarEvents event={event} />, {
+    mocks,
+    path,
+    routes: [route],
+  });
+  for (const similarEvent of expectedSimilarEvents.data) {
+    await act(async () => {
+      await waitForComponentToBeLoaded();
+      userEvent.click(
+        screen.queryByRole('button', {
+          name: translations.event.eventCard.ariaLabelLink.replace(
+            '{{name}}',
+            similarEvent.name.fi
+          ),
+        })
+      );
+      expect(history.location).toMatchObject({
+        pathname: `/fi${ROUTES.EVENT.replace(':id', similarEvent.id)}`,
+        search: `?returnPath=${encodeURIComponent(route)}`,
+      });
+      history.goBack();
+    });
+    await waitForComponentToBeLoaded();
+  }
```

Comment:
usually when using act, you are trying to hide some underlying problem. I tested that this version works without act:

```
  await waitForComponentToBeLoaded();
  for (const similarEvent of expectedSimilarEvents.data) {
      userEvent.click(
        screen.queryByRole('button', {
          name: translations.event.eventCard.ariaLabelLink.replace(
            '{{name}}',
            similarEvent.name.fi
          ),
        })
      );
      expect(history.location).toMatchObject({
        pathname: `/fi${ROUTES.EVENT.replace(':id', similarEvent.id)}`,
        search: `?returnPath=${encodeURIComponent(route)}`,
      });
      history.goBack();
      await waitForComponentToBeLoaded();
  }
```

Reply:
https://kentcdodds.com/blog/fix-the-not-wrapped-in-act-warning

Reply:
Last extra await in the end of the loop was left there without reason, it seem to work without it. Fixed now. Good to know that suggesting using `act` does mean you should use it :)

Reply:
Yes it is little bit confusing but we are actually using act but it is react testing library that is doing it behind the scenes, so most of the time we don't need to write it ourselves

Reply:
"You can read more about React's act utility here, but again, you shouldn't have to use it very often. It's built-into React Testing Library. There are very few times you should have to use it directly if you're using React Testing Library's async utilities."